### PR TITLE
feat: PHP 8 support

### PR DESCRIPTION
**Pull request by ababkov**
# Overview

* Update dependencies to support php 8+
* Force guzzlehttp/psr7 version only because 1.9.x is advertised as okay for guzzlehttp 7.7 but fails in php 8.2

# Before merge

* Merge all "dev-" dependencies, tag release and update appropriately in composer.json

# After merge

Tag as `4.0.0` release

### CODE REVIEW

File: composer
Comment (BlankaK) [inline composer:1]: I don't think I've seen this before. Are we actually including the composer executable in the repo? 🤔

**Review by BlankaK — APPROVED** [review]
I'd love to see a changelog in this library, even if it starts from the current version, I think.
Not quite clear which update specifically requires php 8+ but I have a feeling it's because of the updates made to the underlying libraries.

Approved as I can't see anything wrong with the code, though I am not sure I understand some of the choices made here 😅

File: composer
Comment (ababkov) [reply composer:1]: This was an accident, will remove.

Comment (ababkov) [reply composer:1]: Have removed

**Comment by ababkov** [comment]
> I'd love to see a changelog in this library, even if it starts from the current version, I think. Not quite clear which update specifically requires php 8+ but I have a feeling it's because of the updates made to the underlying libraries.

Also added changelog

**Pull request merged**